### ISSUE TITLE
Error: The API request to CoinGecko was not successful. It returned the following message: Unauthorized

### ISSUE THREAD

**Issue by tadapit** [w3f/staking-rewards-collector#57]
Dear, do you know how to solve to follow error?
Error: The API request to CoinGecko was not successful. It returned the following message: Unauthorized 
kind regards!

**Comment by tadapit** [comment]
I figured out, the free api key can only go 1 year back in time, perhaps mention it in the readme.

**Comment by jonasW3F** [comment]
They must have changed that. Thanks for the info. I'll look into it soon, right now, I just use the public access point without any key. My hope is that it allows calls longer than a year with a free api key. But I need to find the time to look into it. If you have more info, let me know.

**Comment by blockimmo** [comment]
Hi Jonas, any update here? I'm trying to generate some data from last year and the year before, but get the same error. Thank you

**Comment by jonasW3F** [comment]
Yes, I haven't been able to fully review [the PR](https://github.com/w3f/staking-rewards-collector/pull/59) that has been submitted by @rmnprkrl yet, but it works locally for me. It switches to another API provider (cryptocompare). Check it out

**Comment by blockimmo** [comment]
Just did that, works like a charm locally, thank you both @jonasW3F @rmnprkrl

**Comment by jonasW3F** [comment]
The issue is now solved by choosing cryptocompare's API with version 1.7.4
